python/tskit/trees.py

```diff
@@ -3070,6 +3071,124 @@ def simplify(
         else:
             return new_ts
 
+    def slice(
+            self, start=None, stop=None, reset_coordinates=True, simplify=True,
+            record_provenance=True):
+        """
+        Truncate this tree sequence to include only information in the genomic interval
+        between ``start`` and ``stop``.  Edges are truncated to this interval, and sites
+        not covered by the sliced region are thrown away.
+
+        :param float start: The leftmost genomic position, giving the start point
+            of the kept region. Tree sequence information along the genome prior to (but
+            not including) this point will be discarded. If None, set equal to zero.
+        :param float stop: The rightmost genomic position, giving the end point of the
+            kept region. Tree sequence information at this point and further along the
+            genomic sequence will be discarded. If None, is set equal to the current tree
+            sequence's ``sequence_length``.
+        :param bool reset_coordinates: Reset the genomic coordinates such that position
+            0 in the returned tree sequence corresponds to position ``start`` in the
+            original one, and the returned tree sequence has sequence length
+            ``stop``-``start``. Sites and tree intervals will all have their positions
+            shifted to reflect the new coordinate system. If ``False``, do not rescale:
+            the resulting tree sequence will therefore cover the same genomic span as
+            the original, but will have tree and site information missing for genomic
+            regions outside the sliced region. (Default: True)
+        :param bool simplify: If True, simplify the resulting tree sequence so that nodes
+            no longer used in the resulting trees are discarded. (Default: True).
+        :param bool record_provenance: If True, record details of this call to
+            slice in the returned tree sequence's provenance information.
+            (Default: True).
+        :return: The sliced tree sequence.
+        :rtype: .TreeSequence
+        """
+        # First save the args used, to save in provenance
+        func_args, _, _, func_values = inspect.getargvalues(inspect.currentframe())
```

Comment:
This provenance stuff seems like overkill here and a large fraction of the actual code in the method. Can we trim this out and replace it with a simple implementation which doesn't bother with the parameters? Looks like we have a general approach for this elsewhere, so let's do it all at once.

Reply:
OK. I have deleted this and the corresponding tests, but I want to keep the test code around to reintroduce later, so I'm pasting it here, below:

```
    def test_slice_provenance(self):
        ts = msprime.simulate(5, random_seed=1, recombination_rate=2, mutation_rate=2)
        ts_sliced = ts.slice(0.4, 0.6)
        self.assertEqual(ts_sliced.num_provenances, 2)
        provenance = json.loads(ts_sliced.provenance(0).record)
        self.assertEquals(provenance['software']['name'], 'msprime')
        provenance = json.loads(ts_sliced.provenance(1).record)
        self.assertEquals(provenance['parameters']['command'], 'slice')
        self.assertEquals(provenance['parameters']['start'], 0.4)
        self.assertEquals(provenance['parameters']['stop'], 0.6)
        # Check that we don't record provenances if record_provenance=False
        ts_sliced = ts.slice(0.4, 0.6, record_provenance=False)
        self.assertEqual(ts_sliced.num_provenances, 1)
        provenance = json.loads(ts_sliced.provenance(0).record)
        self.assertEquals(provenance['software']['name'], 'msprime')
        # Check we raise an error if we pass something unserializable
        self.assertRaises(TypeError, ts.slice, 0.4, 0.6, record_provenance=lambda x: x)
```